internal/representation/list.go

```diff
@@ -0,0 +1,194 @@
+
+	// ErrUnsupportedContentType indicates an error that occurs when the list
+	// represention is given an unsupported content type.
+	ErrUnsupportedContentType = errors.New("representation content type not supported")
+)
+
+// representation is the core representation.
+type representation struct {
+	encoding      []string
+	mediaType     string
+	charset       string
+	language      string
+	location      url.URL
+	sourceQuality float32
+	features      []string
+}
+
+// ContentType retrieves the content type of the representation.
+func (r representation) ContentType() string { return r.mediaType }
+
+// SetContentType modifies the content type of the representation.
+func (r *representation) SetContentType(ct string) { r.mediaType = ct }
+
+// ContentLanguage retrieves the content language of the representation.
+func (r representation) ContentLanguage() string { return r.language }
+
+// SetContentLanguage modifies the content language of the representation.
+func (r *representation) SetContentLanguage(cl string) { r.language = cl }
+
+// ContentEncoding retrieves the content encoding of the representation.
+func (r representation) ContentEncoding() []string { return r.encoding }
+
+// SetContentEncoding modifies the content encoding of the representation.
+func (r *representation) SetContentEncoding(ce []string) { r.encoding = ce }
+
+// ContentCharset retrieves the content charset of the representation.
+func (r representation) ContentCharset() string { return r.charset }
+
+// SetContentCharset modifies the content charset of the representation.
+func (r *representation) SetContentCharset(cc string) { r.charset = cc }
+
+// ContentLocation retrieves the content location of the representation.
+func (r representation) ContentLocation() url.URL { return r.location }
+
+// SetContentLocation modifies the content location of the representation.
+func (r *representation) SetContentLocation(l url.URL) { r.location = l }
+
+// ContentFeatures retrieves the content features of the representation.
+func (r representation) ContentFeatures() []string { return r.features }
+
+// SourceQuality retrieves the source quality of the representation.
+func (r representation) SourceQuality() float32 { return rep.SourceQualityPerfect }
+
+// SetSourceQuality modifies the source quality of the representation.
+func (r *representation) SetSourceQuality(sq float32) { r.sourceQuality = sq }
+
+// List represents a representation containing a list of descriptions of Representations
+// for a particular resource.
+type List struct {
+	representation
+
+	Representations []rep.Representation
+}
+
+// SetRepresentations modifies the represention list within the list representation.
+func (l *List) SetRepresentations(reps ...rep.Representation) {
+	l.Representations = reps
+}
+
+// Bytes retrieves the serialized form of the list representation.
+func (l List) Bytes() ([]byte, error) {
```

Comment:
should create function on `representation` to consolidate this logic:

```go
func (r representation) Bytes(x interface{}) ([]byte, error) {
    	supportedMediaTypes := map[string]marshaller{
		"application/json": json.Marshal,
		"application/xml":  xml.Marshal,
		"application/yaml": yaml.Marshal,
		"text/yaml":        yaml.Marshal,
	}

	var buf bytes.Buffer
	supportedEncodings := map[string]writerConstructor{
		"gzip":       newGzip,
		"x-gzip":     newGzip,
		"compress":   newCompress,
		"x-compress": newCompress,
		"deflate":    newDeflate,
	}

	if _, ok := supportedMediaTypes[strings.ToLower(r.ContentType())]; !ok {
		return []byte{}, ErrUnsupportedContentType
	}

	// serialize.
	b, err := supportedMediaTypes[strings.ToLower(r.ContentType())](x) //<-----
	if err != nil {
		return b, err
	}

	if len(r.ContentEncoding()) < 1 {
		return b, nil
	}

	// encode.
	var w io.WriteCloser = &closeableBuffer{&buf}
	for _, e := range r.ContentEncoding() {
		if _, ok := supportedEncodings[strings.ToLower(e)]; !ok {
			return []byte{}, ErrUnsupportedContentEncoding
		}
		if w, err = supportedEncodings[strings.ToLower(e)](w); err != nil {
			return []byte{}, err
		}
		defer w.Close()
	}
	if _, err = w.Write(b); err != nil {
		return []byte{}, err
	}

	// return final bytes.
	return buf.Bytes(), nil
}
```
and rewrite this function like so:

```go
// Bytes retrieves the serialized form of the list representation.
func (l List) Bytes() ([]byte, error) {
	return l.Bytes(l)
}
```